dozer-sql/src/pipeline/expression/comparison.rs

```diff
@@ -333,9 +539,14 @@ macro_rules! define_comparison {
                         $op.to_string(),
                     )),
                 },
-                Field::Text(ref left_v) => match right_p {
-                    Field::String(ref right_v) => Ok(Field::Boolean($function(left_v, right_v))),
-                    Field::Text(ref right_v) => Ok(Field::Boolean($function(left_v, right_v))),
+                Field::Date(left_v) => match right_p {
+                    Field::Date(right_v) => Ok(Field::Boolean($function(left_v, right_v))),
+                    Field::String(right_v) | Field::Text(right_v) => {
+                        let right_val = NaiveDate::from_str(right_v.as_str()).map_err(|_| {
+                            PipelineError::UnableToCast(format!("{}", right_v), "Date".to_string())
+                        })?;
```

Comment:
I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

Reply:
fixed